File: cli/tests/golden/artifacts/csharp/customer.CustomerDb.v1.cs
#nullable enable
using System;
using System.Collections.Generic;

namespace Modelable.Customer;

public sealed record CustomerCustomerDbV1
{
    public required Guid CustomerId { get; init; }
    public required string DisplayName { get; init; }
    public required string Email { get; init; }
    public string? InternalRiskNotes { get; init; }
    public required string Status { get; init; }
    public required List<string> Tags { get; init; }
    public required Dictionary<string, int> Metadata { get; init; }
    public CustomerCustomerDbV1Address? Address { get; init; }
    public string? FavoriteProduct { get; init; }
    public required DateTime CreatedAt { get; init; }
    public DateTime? UpdatedAt { get; init; }
}

public sealed record CustomerCustomerDbV1Address
{
    public required string Line1 { get; init; }
    public string? Line2 { get; init; }
}

File: cli/tests/golden/artifacts/typescript/customer.CustomerDb.v1.ts
/**
 * @modelable domain: customer
 * @modelable name: CustomerDb
 * @modelable owner: customer-team
 * @modelable kind: projection
 * @modelable version: 1
 * @modelable source: customer.Customer@1
 */
import type { CatalogProductV1 } from "./catalog.Product.v1";

export interface CustomerCustomerDbV1 {
  customerId: string;
  displayName: string;
  email: string;
  internalRiskNotes?: string;
  status: 'active' | 'suspended' | 'deleted';
  tags: string[];
  metadata: Record<string, number>;
  address?: { line1: string; line2?: string };
  favoriteProduct?: CatalogProductV1;
  createdAt: string;
  updatedAt?: string;
}
export type CustomerDb = CustomerCustomerDbV1;
